### PR TITLE
[KOGITO-972] Allow to change the image-builder when building the operator

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-972

Also added some parameters in Makefile to change the image information, so we can build easily a local image with own registry, name and/or tag.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: hack/go-build.sh [hunk 1]
```diff
@@ -24,12 +41,46 @@ TAG=0.7.0
...
 
 setGoModEnv
 go generate ./...
 if [[ -z ${CI} ]]; then
     ./hack/go-test.sh
-    operator-sdk build ${REGISTRY}/${IMAGE}:${TAG}
+    BUILD_PARAMS=""
+    if [[ ! -z ${BUILDER} ]]; then BUILD_PARAMS="${BUILD_PARAMS} --image-builder ${BUILDER}"; fi
+    operator-sdk build ${REGISTRY}/${IMAGE}:${TAG} ${BUILD_PARAMS}
     if [[ ${1} == "rhel" ]]; then
```
Comment (ricardozanini) [inline hack/go-build.sh]: I think we can drop this, let's take care of productization once we have the Cekit enabled.

Comment (spolti) [reply hack/go-build.sh]: not sure what would be better, still with cekit or use operator-sdk build.
Maybe have both enabled?

Comment (ricardozanini) [reply hack/go-build.sh]: I'd rather going for Cekit, like we are doing on downstream, or maybe a `operator-sdk` plugin for cekit :thinking:

Comment (spolti) [reply hack/go-build.sh]: I don't think cekit will have operator-sdk plugin.
but let's use cekit only for now.

Comment (spolti) [reply hack/go-build.sh]: what about leave it as it is and update once we migrate to cekit?

**Comment by radtriste** [comment]
referencing the issue for Cekit migration: https://issues.redhat.com/browse/KOGITO-233

**Pull request merged**